overlord/configstate/config/transaction.go

```diff
@@ -343,6 +404,69 @@ func commitChange(pristine *json.RawMessage, change interface{}) *json.RawMessag
 	panic(fmt.Errorf("internal error: unexpected configuration type %T", change))
 }
 
+// mergeConfigWithVirtual takes the given configuration and merges it
+// with the virtual configuration values by calling the registered
+// virtual configuration function of the given snap. The merged config
+// is returned.
+func mergeConfigWithVirtual(instanceName string, config *json.RawMessage) (*json.RawMessage, error) {
+	virtualMu.Lock()
+	km, ok := virtualMap[instanceName]
+	virtualMu.Unlock()
```

Comment:
Should this be defer virtualMu.Unlock() to keep the map locked for the duration of this function?

Reply:
I think that is ok, just locking briefly, the functions should run in parallel, it's really just the (global) map access that I want to serialize. Let me know if I'm missing something here.